packages/reactivity/src/ref.ts

```diff
@@ -72,11 +72,13 @@ function toProxyRef<T extends object, K extends keyof T>(
   }
 }
 
+type UnwrapArray<T> = { [P in keyof T]: UnwrapRef<T[P]> }
+
 // Recursively unwraps nested value bindings.
 export type UnwrapRef<T> = {
   cRef: T extends ComputedRef<infer V> ? UnwrapRef<V> : T
   ref: T extends Ref<infer V> ? UnwrapRef<V> : T
-  array: T extends Array<infer V> ? Array<UnwrapRef<V>> : T
+  array: T extends Array<infer V> ? Array<UnwrapRef<V>> & UnwrapArray<T> : T
```

Comment:
I think this can simply be `array: T extends Array<any> ? { [K in keyof T]: UnwrapRef<T[K]> } : T`

Reply:
My first commit was similar to that, but for some reason it fails: https://app.circleci.com/jobs/github/vuejs/vue-next/1222

For what I've found the `children.value` doesn't get recognised as a valid array
https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/__tests__/rendererPortal.spec.ts#L68